### PR TITLE
Configure GCS bucket for terraform state.

### PR DESCRIPTION
**Pull request by jiggoha**
https://cloud.google.com/docs/terraform/resource-management/store-state

### CODE REVIEW

File: deployment/main.tf [hunk 1]
```diff
@@ -329,4 +338,32 @@ resource "google_kms_key_ring" "firmware_release_prod" {
 #   }
 # }
 
+resource "google_kms_key_ring" "terraform_state" {
+  name = "armored-witness-bucket-tfstate"
+  location = "europe-west2"
```
Comment (AlCutter) [inline deployment/main.tf:343]: would it be a good idea to make this into a `var.state_location` and reuse in the `google_storeage_bucket` below too? (just thinking that if we decided to move state elsewhere, but accidentally only moved one of these two things, we'd be doubling our exposure to unavailability risk)

File: deployment/main.tf [hunk 2]
```diff
@@ -329,4 +338,32 @@ resource "google_kms_key_ring" "firmware_release_prod" {
...
+}
+
+resource "google_kms_crypto_key" "terraform_state_bucket" {
+  name            = "terraform-state-bucket"
+  key_ring        = google_kms_key_ring.terraform_state.id
+  rotation_period = "86400s"
+
+  lifecycle {
```
Comment (AlCutter) [inline deployment/main.tf:351]: Just wanted to ask whether this was intended to stay in here, or if it's just during dev?

**Review by AlCutter — APPROVED** [review]

File: deployment/main.tf (hunk 1 above)
Comment (jiggoha) [reply deployment/main.tf]: Done.

File: deployment/main.tf (hunk 2 above)
Comment (jiggoha) [reply deployment/main.tf]: I got it from the doc, but looks like it's unnecessary. Removed.

**Pull request merged**